zenoh-ext/src/querying_subscriber.rs

```diff
@@ -672,9 +672,10 @@ impl<'a, Handler> FetchingSubscriber<'a, Handler> {
                     tracing::trace!(
                         "Sample received while fetch in progress: push it to merge_queue"
                     );
+
                     // ensure the sample has a timestamp, thus it will always be sorted into the MergeQueue
                     // after any timestamped Sample possibly coming from a fetch reply.
-                    let timestamp = s.timestamp().cloned().unwrap_or(new_timestamp(zid));
+                    let timestamp = s.timestamp().cloned().unwrap_or(timestamp);
```

Comment:
Same problem than in `plugins/zenoh-plugin-storage-manager/src/replica/storage.rs` :
a new timestamp shall be generated for each sample received without timestamp

Reply:
In the case of Querying Subscriber,
inside `conf` we get a session of type `session: SessionRef<'a>`
```rust
#[derive(Clone)]
pub enum SessionRef<'a> {
    Borrow(&'a Session),
    Shared(Arc<Session>),
}
```
In the case of it being a `Shared` it can be cloned and passed off to the callback. 
In the case of it being `Borrow` the callback's lifetime will need to be tied to the `SessionRef`

I think its a cleaner for both cases to just take the `ZenohID` and generate a timestamp inside the callback.